### PR TITLE
nvOC as systemd service

### PR DESCRIPTION
**Pull request by LuKePicci**
As @doftorul suggested in #41 , here is an attempt to get rid of gnome-terminal by using systemd.

This requires a NOPASSWD sudoer, nologin user called `nvoc`. You could instead run as user `m1` (default one in nvOC OS images) which still has this requirement satisfied by default. Just tweak the `User=` option (systemd/system) `nvoc.service`.

Known issues:
- `DISPLAY=:0` is missing in tempcontrol (at least), therefore fan setting is failing
-- ✔Fixed: there was a problem accessing X server cookie
- Working file like those used by switchers need to be saved somewhere with write access
-- ✔Fixed: granted write access to nvoc user for ${NVOC} installation directory

Note:
- Logs are not written in `tmpfs` ramdrive before it is mounted or if mounting problems occur, so If you need to check `main-screenlog.0` for errors occurred before ramdrive mounting (which happens very early in 2unix), unmount that ramdrive first and look for the real content of the `nvoc_logs` directory

### CODE REVIEW

**Comment by papampi** [comment]
2 small problems ...

**Comment by LuKePicci** [comment]
A preview:
image

**Comment by papampi** [comment]
Very interesting ...

File: 3main [hunk 1]
```diff
@@ -159,23 +159,24 @@ NORMAL="NO"
 
 P10X_100="NO"
 
-nvidia-smi -L > /tmp/tempa
+if [ ! -f ${NVOC}/nvoc_logs/tempa ]
```
Comment (papampi) [inline 3main]: Good one ...

**Comment by LuKePicci** [comment]
@doftorul if you don't use any auto switcher you may want to give this PR a look already.

**Review by papampi — DISMISSED** [review]

**Comment by papampi** [comment]
I'm out of town and can't test, please merge after checked.

**Comment by LuKePicci** [comment]
This one is still WIP, will do it later on this week.

Inviato dal mio telefono Windows 10

Da: Payam Nab
Inviato: mercoledì 30 maggio 2018 16:27
A: papampi/nvOC_by_fullzero_Community_Release
Cc: Luca Piccirillo; Author
Oggetto: Re: [papampi/nvOC_by_fullzero_Community_Release] nvOC as systemdservice (#62)

I'm out of town and can't test, please merge after checked.

**Comment by LuKePicci** [comment]
Ok, I'm happy to inform you that this PR is ready, the issue with writing permission for whatever needs to write files in the NVOC directory is best solved by setting such permission to the service user. I did so in the service install script, now _should_ work.

**Review by papampi — APPROVED** [review]
@LuKePicci I dont have time to test now, please test yourself and merge.

**Pull request merged**